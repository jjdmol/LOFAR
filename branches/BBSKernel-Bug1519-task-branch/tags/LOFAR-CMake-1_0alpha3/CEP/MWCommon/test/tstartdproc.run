#!/bin/sh

# tstartdproc: Test script startdistproc in various ways
#
# Copyright (C) 2008
# ASTRON (Netherlands Institute for Radio Astronomy)
# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
#
# This file is part of the LOFAR software suite.
# The LOFAR software suite is free software: you can redistribute it and/or
# modify it under the terms of the GNU General Public License as published
# by the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# The LOFAR software suite is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
#
# @author Ger van Diepen <diepen AT astron nl>
#
# $Id$


if test "$srcdir" = ""; then
  srcdir=../../../src
fi

echo ''
$srcdir/../src/startdistproc -dsn tstartdproc.in_vd -cdn tstartdproc.in_cd \
-mode single -dry -hfn tstartdproc_tmp.out1 -fdp ../src prog1 -arg "a b"
cat tstartdproc_tmp.out1

echo ''
$srcdir/../src/startdistproc -dsn tstartdproc.in_vd -cdn tstartdproc.in_cd \
-mode mpi -dry -hfn tstartdproc_tmp.out2 -fdp ../src prog1 -arg "a b"
cat tstartdproc_tmp.out2

echo ''
$srcdir/../src/startdistproc -useenv tstartdproc_tmp.out3.envfile -dsn tstartdproc.in_vd -cdn tstartdproc.in_cd -mode 3851 -dry -hfn tstartdproc_tmp.out3 -fdp ../src prog1 -arg "a b"
cat tstartdproc_tmp.out3

