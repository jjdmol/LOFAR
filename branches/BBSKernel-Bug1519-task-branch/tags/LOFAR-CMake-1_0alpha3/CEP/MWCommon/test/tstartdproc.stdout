
prog1 -arg a b
localhost
node1#/usr/local/xyx0,node1:/usr,/usr/local/xyx0.vds
node2#/usr/local/xyx1,node1:/usr,/usr/local/xyx1.vds
node1#/usr/local/xyx2,node1:/usr,/usr/local/xyx2.vds

mpirun -np 4 -machinefile tstartdproc_tmp.out2 prog1  -arg a b
localhost
node1
node2
node1

socketrun 1 tstartdproc_tmp.out3 'localhost' 3851 1  prog1 tstartdproc_tmp.out3.envfile -arg a b
ssh -x -n localhost prog1 socket localhost 3851 4 0 '.' '.' '.'  'tstartdproc_tmp.out3.envfile' '-arg' 'a b'  &
ssh -x -n node1 prog1 socket localhost 3851 4 1 '/usr/local/xyx0' 'node1:/usr' '/usr/local/xyx0.vds'  'tstartdproc_tmp.out3.envfile' '-arg' 'a b'  &
ssh -x -n node2 prog1 socket localhost 3851 4 2 '/usr/local/xyx1' 'node1:/usr' '/usr/local/xyx1.vds'  'tstartdproc_tmp.out3.envfile' '-arg' 'a b'  &
ssh -x -n node1 prog1 socket localhost 3851 4 3 '/usr/local/xyx2' 'node1:/usr' '/usr/local/xyx2.vds'  'tstartdproc_tmp.out3.envfile' '-arg' 'a b'  &
localhost
node1#/usr/local/xyx0,node1:/usr,/usr/local/xyx0.vds
node2#/usr/local/xyx1,node1:/usr,/usr/local/xyx1.vds
node1#/usr/local/xyx2,node1:/usr,/usr/local/xyx2.vds
